extensions/common/json-ld/src/test/java/org/eclipse/edc/jsonld/transformer/to/JsonObjectToConstraintTransformerTest.java

```diff
@@ -57,49 +70,157 @@ void setUp() {
+                                .add(VALUE, operator.name())))
+                .add(ODRL_RIGHT_OPERAND_ATTRIBUTE, jsonFactory.createArrayBuilder()
+                        .add(jsonFactory.createObjectBuilder()
+                                .add(VALUE, right)))
+                .build();
+
+        var result = transformer.transform(constraint, context);
+
+        assertThat(result).isNotNull().asInstanceOf(type(AtomicConstraint.class)).satisfies(atomicConstraint -> {
+            assertThat(((LiteralExpression) atomicConstraint.getLeftExpression()).getValue()).isEqualTo(left);
+            assertThat(atomicConstraint.getOperator()).isEqualTo(operator);
+            assertThat(((LiteralExpression) atomicConstraint.getRightExpression()).getValue()).isEqualTo(right);
+        });
+        verifyNoInteractions(context);
+    }
+
+    @Test
+    void atomicConstraint_operatorAsString_returnConstraint() {
+        var left = "left";
+        var operator = Operator.EQ;
+        var right = "right";
+
+        var constraint = jsonFactory.createObjectBuilder()
+                .add(ODRL_LEFT_OPERAND_ATTRIBUTE, jsonFactory.createObjectBuilder()
+                        .add(VALUE, left))
+                .add(ODRL_OPERATOR_ATTRIBUTE, operator.name())
+                .add(ODRL_RIGHT_OPERAND_ATTRIBUTE, jsonFactory.createObjectBuilder()
+                        .add(VALUE, right))
                 .build();
-        var andConstraint = AndConstraint.Builder.newInstance().build();
 
-        when(context.transform(isA(JsonObject.class), eq(MultiplicityConstraint.class))).thenReturn(andConstraint);
+        var result = transformer.transform(constraint, context);
 
-        var result = transformer.transform(getExpanded(logicalConstraintJson), context);
+        assertThat(result).isNotNull().asInstanceOf(type(AtomicConstraint.class))
+                .extracting(AtomicConstraint::getOperator)
+                .isEqualTo(operator);
 
-        assertThat(result).isEqualTo(andConstraint);
-        verify(context, times(1)).transform(isA(JsonObject.class), eq(MultiplicityConstraint.class));
+        verifyNoInteractions(context);
     }
 
     @Test
-    void transform_shouldReportProblem_whenUnknownConstraint() {
-        var invalidConstraintJson = jsonFactory.createObjectBuilder()
-                .add(JsonLdKeywords.CONTEXT, JsonObject.EMPTY_JSON_OBJECT)
-                .add(ODRL_SCHEMA + "invalid", "invalid")
+    void atomicConstraint_invalidAttributes_reportProblems() {
+        var jsonNumber = Json.createValue(1);
+
+        var constraint = jsonFactory.createObjectBuilder()
+                .add(ODRL_LEFT_OPERAND_ATTRIBUTE, jsonNumber)
+                .add(ODRL_OPERATOR_ATTRIBUTE, jsonNumber)
+                .add(ODRL_RIGHT_OPERAND_ATTRIBUTE, jsonNumber)
                 .build();
 
-        var result = transformer.transform(getExpanded(invalidConstraintJson), context);
+        var result = transformer.transform(constraint, context);
```

Comment:
shouldn't this be expanded first to always ensure the input?

Reply:
I wasn't aware that this need to be done on all the tests. Did it